controllers/k8sgpt_controller.go

```diff
@@ -218,6 +223,52 @@ func (r *K8sGPTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 					Namespace: k8sgptConfig.Namespace,
 				},
 			}
+			if k8sgptConfig.Spec.Backstage {
+				labelKey := "backstage.io/kubernetes-id"
+				namespace, resourceName, _ := strings.Cut(resultSpec.Name, "/")
+				m, err := cmdutil.NewFactory(genericclioptions.NewConfigFlags(true)).ToRESTMapper()
+				if err != nil {
+					k8sgptReconcileErrorCount.Inc()
+					return r.finishReconcile(err, false)
+				}
+
+				gvr, err := m.ResourceFor(schema.GroupVersionResource{
+					Resource: resultSpec.Kind,
+				})
+				if err != nil {
+					k8sgptReconcileErrorCount.Inc()
+					return r.finishReconcile(err, false)
+				}
+
+				obj := &unstructured.Unstructured{}
+				gvk := schema.GroupVersionKind{
+					Group:   gvr.Group,
+					Kind:    resultSpec.Kind,
+					Version: gvr.Version,
+				}
+
+				obj.SetGroupVersionKind(gvk)
+
+				// Retrieve the resource using the client
+				err = r.Client.Get(ctx, client.ObjectKey{Name: resourceName, Namespace: namespace}, obj)
+				if err != nil {
+					if errors.IsNotFound(err) {
+						k8sgptReconcileErrorCount.Inc()
+						return r.finishReconcile(err, false)
+					} else {
+						k8sgptReconcileErrorCount.Inc()
+						return r.finishReconcile(err, false)
+					}
+				}
+				labels := obj.GetLabels()
+				if value, exists := labels[labelKey]; exists {
+					// Assign the same label key/value to result CR
+					result.ObjectMeta.Labels = map[string]string{labelKey: value}
+				} else {
+					// too verbose?
+					fmt.Printf("Label key '%s' does not exist in %s resource: %s\n", labelKey, resultSpec.Kind, obj.GetName())
```

Comment:
This is a small detail but it probably makes sense to use the `resourceName`

Reply:
done